ts/test/test.oauth2.ts

```diff
@@ -931,6 +931,26 @@ describe('OAuth2 client', () => {
       });
     });
 
+    it('should refresh if access token will expired soon and time to refresh' +
+           ' before expiration is set',
+       (done) => {
+         const auth = new GoogleAuth();
+         const oauth2client = new auth.OAuth2(
+             CLIENT_ID, CLIENT_SECRET, REDIRECT_URI, undefined, 5000);
+
+         oauth2client.credentials = {
+           access_token: 'initial-access-token',
+           refresh_token: 'refresh-token-placeholder',
+           expiry_date: (new Date()).getTime() + 3000
+         };
+
+         oauth2client.request({uri: 'http://example.com'}, () => {
```

Comment:
I hate to do this... but it probably makes more sense to rebase this PR against the `next` branch.